File: articles/algorithmic-trust-verification-and-security/calculators/r/algorithmic_trust_calculator.R
# algorithmic_trust_calculator.R
args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)
if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  root <- normalizePath(file.path(dirname(script_path), "..", ".."), mustWork = TRUE)
} else {
  root <- getwd()
}

out_dir <- file.path(root, "calculators", "outputs")
if (!dir.exists(out_dir)) dir.create(out_dir, recursive = TRUE)

trust_quality <- function(verification, validation, security, provenance, monitoring, governance) {
  100 * (0.18 * verification + 0.18 * validation + 0.18 * security + 0.16 * provenance + 0.15 * monitoring + 0.15 * governance)
}

residual_risk <- function(initial_likelihood, initial_impact, control_strength) {
  100 * initial_likelihood * initial_impact * (1 - control_strength)
}

calibration_gap <- function(human_reliance, system_reliability) {
  abs(human_reliance - system_reliability)
}

result <- data.frame(
  trust_quality_score = trust_quality(0.88, 0.82, 0.88, 0.90, 0.84, 0.82),
  residual_risk_score = residual_risk(0.70, 0.75, 0.62),
  calibration_gap = calibration_gap(0.88, 0.72)
)

write.csv(result, file.path(out_dir, "algorithmic_trust_calculator_r.csv"), row.names = FALSE)
print(result)

File: articles/algorithmic-trust-verification-and-security/calculators/python/algorithmic_trust_calculator.py
# algorithmic_trust_calculator.py
from __future__ import annotations

from pathlib import Path
import json

ROOT = Path(__file__).resolve().parents[2]
OUT = ROOT / "calculators" / "outputs"


def trust_quality(verification: float, validation: float, security: float, provenance: float, monitoring: float, governance: float) -> float:
    return 100.0 * (0.18 * verification + 0.18 * validation + 0.18 * security + 0.16 * provenance + 0.15 * monitoring + 0.15 * governance)


def residual_risk(initial_likelihood: float, initial_impact: float, control_strength: float) -> float:
    return 100.0 * initial_likelihood * initial_impact * (1.0 - control_strength)


def calibration_gap(human_reliance: float, system_reliability: float) -> float:
    return abs(human_reliance - system_reliability)


def main() -> None:
    payload = {
        "trust_quality_score": round(trust_quality(0.88, 0.82, 0.88, 0.90, 0.84, 0.82), 3),
        "residual_risk_score": round(residual_risk(0.70, 0.75, 0.62), 3),
        "calibration_gap": round(calibration_gap(0.88, 0.72), 3),
        "interpretation": "Trust quality combines verification, validation, security, provenance, monitoring, and governance; residual risk and calibration gaps still require review."
    }
    OUT.mkdir(parents=True, exist_ok=True)
    (OUT / "algorithmic_trust_calculator_python.json").write_text(json.dumps(payload, indent=2), encoding="utf-8")
    print(json.dumps(payload, indent=2))


if __name__ == "__main__":
    main()
